wcfsetup/install/files/js/WCF.Attachment.js

```diff
@@ -308,6 +315,85 @@ if (COMPILER_TARGET_DEFAULT) {
 			// remove failed uploads
 			this._fileListSelector.children('li.uploadFailed').remove();
 			
+			if (this._options.enableAutoScale) {
+				var self = this;
+				var maxSize = this._buttonSelector.data('maxSize');
+				
+				var promise = new Promise(function (resolve) {
+					require(['ImageResizer'], function (ImageResizer) {
+						var promises = [];
+						
+						// Create an ImageResizer instance
+						var resizer = new ImageResizer()
+							.setMaxWidth(self._options.autoScaleMaxWidth)
+							.setMaxHeight(self._options.autoScaleMaxHeight)
+							.setFileType(self._options.autoScaleFileType)
+							.setQuality(self._options.autoScaleQuality);
+						
+						for (var i = 0; i < files.length; i++) {
+							if (files[i].type.match(/^image\//i)) {
+								(function (i) {
```

Comment:
This IIFE is redundant, because there is a second closure (the parameter of the `Promise` constructor).